pkgs/development/python-modules/django-filter/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib
+, fetchPypi
+, django
+, buildPythonPackage
+}:
+buildPythonPackage rec {
+
+  pname = "django-filter";
+  version = "2.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "3dafb7d2810790498895c22a1f31b2375795910680ac9c1432821cbedb1e176d";
+  };
+
+  doCheck = false;
```

Comment:
It's best to leave a reason for disabling tests. In this case the tests seem to require `crispy_forms` which isn't in nixpkgs (yet).

Reply:
Oh hold on https://github.com/NixOS/nixpkgs/pull/63502

Didn't see that. It might be better to combine this series of PRs into one...

Reply:
Ok provided you have `django-crispy-forms` available, it's quite simple to enable the tests:

```
  checkInputs = [
    djangorestframework
    mock
    django-crispy-forms
  ];
  checkPhase = ''
    ${python.interpreter} runtests.py
  '';
```

However a single test fails. This corresponds to an upstream issue https://github.com/carltongibson/django-filter/issues/1050 which has luckily been fixed, and we can pull in that changeset as a patch:

```
  patches = [
    (fetchpatch {
      name = "fix-dst-test.patch";
      url = https://github.com/carltongibson/django-filter/commit/9c0b06610dbd5b69a006ae121d383ef49cc70fff.patch;
      sha256 = "0rrbqxjcwwkcldxi1q38cf2wcl9rgk7yyyc6rxwswi45r9sxihgb";
    })
  ];
```

With the above two modifications, the tests pass for me non-nixos linux x86_64

Reply:
Thank you!
Yeah I have a slew of PRs from a big collection of modules I added at work.
I want to get them all contributed but I'm pretty unfamiliar with the "norms" of contributing (despite reading the guide several times).

Maybe you can clear this up: is it best for me to submit one big PR with all of my new modules? (About 6 or 7 Python modules). Alternatively should I make PRs 1 at a time while I wait for each dependency in the chain to be merged?
What is the deal with the maintainers list? I have tried to add myself on other PRs for like a year now and every time they tell me to remove it. Am I supposed to add myself as a maintainer on a separate PR? When I tried that before it was also rejected for some reason... Is the idea that I just push code anonymously until I have a fat collection of packages that I finally "claim" from my past PRs?

I appreciate any advice you can spare.

Reply:
Certainly if they're interrelated, I'd combine your PRs together.

The maintainers list? I don't know - I never had any issue adding myself, in fact I was nagged for creating packages and leaving them maintainerless. I think there's probably a stronger case to add yourself to the maintainers list if you're adding a slew of packages together which you're maintaining.

When it comes to getting tests working, I have two main tricks/pieces of advice:
1. Many pypi packages don't include the test suite, in this case you can pull the source directly from the repo with e.g. `fetchFromGitHub`.
2. A great source for "how am I supposed to get the tests running from scratch for this project?" is looking at the project's CI configuration. `.travis.yml`s, `tox.ini`s, `appveyor.yml`s etc.

Reply:
(the tests for `django-crispy-forms` are looking pretty tricky and might be beyond hope)

Reply:
> I have tried to add myself on other PRs for like a year now and every time they tell me to remove it

Just a thought - do you think it might be something to do with your github username?